cmd/minikube/cmd/service.go

```diff
@@ -67,3 +78,35 @@ func init() {
 	serviceCmd.Flags().BoolVar(&serviceURLMode, "url", false, "Display the kubernetes service URL in the CLI instead of opening it in the default browser")
 	RootCmd.AddCommand(serviceCmd)
 }
+
+// CheckService waits for the specified service to be ready by returning an error until the service is up
```

Comment:
Could you make it clear that this returns a function that checks the service?

Actually, it might be easier to just make this do the checking. Then your retry calls could use an anonymous function.

commonutil.RetryAfter(20, func() error { return CheckService(namespace, service)})